libnmstate/schemas/operational-state.yaml

```diff
@@ -240,6 +261,12 @@ definitions:
                       type: integer
                     max-age:
                       type: integer
+                vlan-filtering:
```

Comment:
We can re-use this definition for the uplink / self bridge port. 

I see no reason at all to come up w/ yet another representation.

Reply:
I'm *just* providing a way for the user to disable vlan filtering altogether. I think it's useful.

Do you need an example for that ? 

Furthermore, if you really want the uplink port defined, we can set the vlan port definition also here; it would apply to the uplink / self port .

Reply:
> I'm just providing a way for the user to disable vlan filtering altogether. I think it's useful.

I'm asking about the two level part, not about the need to have this enable/disable option.
Looking for the reason this simpler options is not enough: `vlan-filtering: False`

> Do you need an example for that ?

I need an example of a future property that you think may fit under this hierarchy (therefore, explaining its need).

> Furthermore, if you really want the uplink port defined, we can set the vlan port definition also here; it would apply to the uplink / self port .

We had two options, one to set it perhaps here or flatted under options, or mention it in the port list. If you prefer the first, then it could be the example I was looking for.

Reply:
> > I'm just providing a way for the user to disable vlan filtering altogether. I think it's useful.
> 
> I'm asking about the two level part, not about the need to have this enable/disable option.
> Looking for the reason this simpler options is not enough: `vlan-filtering: False`

Oh, I see. From past conversation w/ @phoracek, we chose the option of having it as a map to possibly incorporate further data - e.g. the 'self' port vlan filtering configuration.

Choosing to model that port as 'just another port' would indeed lessen the requirement for having that property within a map.

It also feels more in line with what's on STP, for instance (again, **only if** there are more attributes).

> 
> > Do you need an example for that ?
> 
> I need an example of a future property that you think may fit under this hierarchy (therefore, explaining its need).

Roger that.

> 
> > Furthermore, if you really want the uplink port defined, we can set the vlan port definition also here; it would apply to the uplink / self port .
> 
> We had two options, one to set it perhaps here or flatted under options, or mention it in the port list. If you prefer the first, then it could be the example I was looking for.

Will consider both options, then come up with a proposal. Too much uncertainty at the moment.

Reply:
Ok, so lets not implement this here as you said. We can interpret if it is enabled or not for now.
If vlans are set, then this is implicitly enabled, otherwise, it is disabled. What do you think?